File: app/src/main/java/com/lzh/nonview/router/demo/LoginActivity.java
package com.lzh.nonview.router.demo;

import android.net.Uri;
import android.os.Bundle;

import com.lzh.compiler.parceler.annotation.Arg;
import com.lzh.nonview.router.Router;
import com.lzh.nonview.router.demo.base.BaseActivity;
import com.lzh.nonview.router.route.ActivityRouteBundleExtras;

import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.Extra;

import butterknife.OnClick;

public class LoginActivity extends BaseActivity {

    @Arg
    Uri uri;
    @Arg
    ActivityRouteBundleExtras extras;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
    }

    @OnClick(R.id.login)
    void onLoginClick () {
        DataManager.INSTANCE.setLogin(true);
        Router.create(uri).getActivityRoute().replaceBundleExtras(extras).open(this);
        finish();
    }
}

File: app/src/main/java/com/lzh/nonview/router/demo/LoginActivity.kt
package com.lzh.nonview.router.demo

import android.net.Uri
import android.os.Bundle
import butterknife.OnClick
import com.lzh.compiler.parceler.annotation.Arg
import com.lzh.nonview.router.Router
import com.lzh.nonview.router.demo.manager.DataManager
import com.lzh.nonview.router.extras.RouteBundleExtras

class LoginActivity:BaseActivity() {

    @Arg("uri")
    var uri:Uri? = null
    @Arg("extras")
    var extras:RouteBundleExtras? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
    }

    @OnClick(R.id.login)
    fun login() {
        DataManager.login = true
        // 登录完成后恢复路由启动。
        Router.resume(uri, extras).open(this)
        finish()
    }
}

